dra-evolution/pkg/api/capacity_types.go

```diff
@@ -69,28 +74,94 @@ type ResourcePoolSpec struct {
 	// them) empty pool.
 }
 
+const ResourcePoolMaxSharedCapacity = 128
 const ResourcePoolMaxDevices = 128
 
+type DeviceShape struct {
```

Comment:
I'm still struggling to see how I share attributes across top level GPUs. Or is that not supported in this model? Or do I need to put all of my top-level GPUs into one DeviceShape to take advantage of this?

Reply:
There is exactly one shape for any slice. Check out the example YAML. 

Cardinality:

- One slice has one shape. The slice effectively encapsulates a model of device. If you have a node with several models of GPU, you would have a slice for each model.
- One shape has many partitions. This is the "map" of how this shape device can be partitioned. 
- One slice has many devices, all of the same, single shape. Thus, the only thing you need in device entries are the device names and any device-specific attributes. 



What you cannot do in this as it exists here is have per-device, per-partition attribute. But otherwise it is as simple expressive as the fully denormalized model.

If we want to pack things in tighter, we could allow multiple shapes per slice, but I don't see any value in that.

Reply:
I need each MIG device to have a reference to its Parent GPU's UUID (so that they can be aligned on that). How does one do that with this model? Likewise, for PCIEroot, etc.

Reply:
The device as seen by the scheduler will have all the attributes of the shape, the partition, and the device. So the device UUID will be there for every MIG.

Now, if we need to have a separate UUID for the MIG device and for the parent device, it is a bit more tricky, because the we may have name conflicts. 

So, we may need a little more attribute manipulation magic here. And we may need a standard attribute to mean "partition". I prefer to see if we can get by with published conventions rather than manipulations. Another alternative is qualifying attributes by shape, partition, and device, but that seems very hard on the user.

Reply:
> The device as seen by the scheduler will have all the attributes of the shape, the partition, and the device. So the device UUID will be there for every MIG.

The idea is that the scheduler sees an unrolled / flattened / projected version of this into a flat list of devices. So from the claim standpoint, it is equivalent to the current model.